src/libraries/System.Private.CoreLib/src/System/Marvin.OrdinalIgnoreCase.cs

```diff
@@ -48,10 +48,26 @@ public static int ComputeHash32OrdinalIgnoreCase(ref char data, int count, uint
                     goto NotAscii;
                 }
 
-                // addition is written with -0x80u to allow fall-through to next statement rather than jmp past it
-                p0 += Utf16Utility.ConvertAllAsciiCharsInUInt32ToUppercase(tempValue) + (0x800000u - 0x80u);
+                if (BitConverter.IsLittleEndian)
+                {
+                    p0 += Utf16Utility.ConvertAllAsciiCharsInUInt32ToUppercase(tempValue) + 0x800000u;
+                }
+                else
+                {
+                    p0 += (Utf16Utility.ConvertAllAsciiCharsInUInt32ToUppercase(tempValue) << 16) + 0x8000u;
+                }
+            }
+            else
+            {
+                if (BitConverter.IsLittleEndian)
+                {
+                    p0 += 0x80u;
```

Comment:
This shouldn't be in an 'else' block. (See the comment on line 51 in the baseline file.)

Reply:
@GrabYourPitchforks I've now kept that micro-optimization on little-endian as-is, and also added it to the big-endian case.  This is a bit awkward since on big-endian the optimization may cause spurious integer overflows, so I had to add "unchecked" blocks.

(I had originally simply removed that micro-optimization since, unless I'm missing something here, on current CPU microarchitectures it seems no longer really useful anyway - an extra unconditional branch should likely be faster than an extra addition.)